### PR TITLE
doc/dev: edit delayed-delete.rst

### PR DESCRIPTION
**Pull request by zdover23**
This PR improves the English in the "CephFS Delayed
Deletion" chapter of the Ceph Internals Guide (which
is stored in the doc/dev/ directory).

There was no pressing need to rewrite this. I just saw
that it could be tightened up, and I had the fifteen
minutes I needed to do it.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/dev/delayed-delete.rst [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
-``size_of_file / stripe_size * replication_count`` messages would slow
-the client down too much, and use a too much of the clients
-bandwidth. Additionally, snapshots may mean some objects should not be
-deleted.
+The deletion of a file does not immediately remove its data. Each
+object in the file must be removed independently. Sending
+``size_of_file / stripe_size * replication_count`` messages slows
+the client unacceptably, and consumes too much client bandwidth. 
+If snapshots exist, their existence can prevent the deletion of 
+objects associated with them.
```
Comment (anthonyeleven) [inline doc/dev/delayed-delete.rst:10]: If I understand the intent of these sentences, maybe something like

Each of the file's underlying objects must also be explicitly removed.  Doing so immediately would require the client to send ``size_of_file / stripe_size * replication_count`` messages to the Ceph cluster, which would consume significant bandwidth and slow the client unacceptably.

File: doc/dev/delayed-delete.rst [hunk 2]
```diff
@@ -2,11 +2,12 @@
...
 
-Instead, the file is marked as deleted on the MDS, and deleted lazily.
+In these cases, such files are (1) marked as deleted on the MDS and 
+(2) are deleted lazily.
```
Comment (anthonyeleven) [inline doc/dev/delayed-delete.rst:13]: With the first `are` preceding `(1)`, the second following `(2)` seems non-congruent.  I might suggest "(2) deleted lazily", that is, remove the second `are`.

**Review by anthonyeleven — APPROVED** [review]

File: doc/dev/delayed-delete.rst (hunk 2 above)
Comment (zdover23) [reply doc/dev/delayed-delete.rst:13]: I see your point.

**Comment by anthonyeleven** [comment]
Lgtm

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/47051 - Quincy backport
https://github.com/ceph/ceph/pull/47050 - Pacific backport

**Pull request merged**